module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -746,30 +746,21 @@ function ($a, $b) use ($sortField) {
      *
      * @param object $item The item record
      *
-     * @return array An array of key metadtaa for each bib record
+     * @return array An array of key metadata for each bib record
      */
     protected function getBoundWithRecords($item)
     {
         $boundWithRecords = [];
-        // Get the bound-with holdings for the item.
-        foreach (
-            $this->getPagedResults(
-                'boundWithParts',
-                '/inventory-storage/bound-with-parts',
-                ['query' => '(itemId=="' . $item->id . '")']
-            ) as $boundWithPart
-        ) {
-            $boundWithHoldingId = $boundWithPart->holdingsRecordId;
-            $response = $this->makeRequest(
-                'GET',
-                '/holdings-storage/holdings/' . $boundWithHoldingId
-            );
-            $holding = json_decode($response->getBody());
-            // Get the bound-with holding's instance record.
-            $instance = $this->getInstanceById($holding->instanceId);
+        // Get the full item record, which includes the boundWithTitles data
+        $response = $this->makeRequest(
+            'GET',
+            '/inventory/items/' . $item->id
+        );
+        $item = json_decode($response->getBody());
+        foreach ($item->boundWithTitles ?? [] as $boundWithTitle) {
             $boundWithRecords[] = [
-                'title' => $instance->title,
-                'bibId' => $this->getBibId($instance),
+                'title' => $boundWithTitle->briefInstance?->title,
```

Comment:
Is it okay for the title value to be null? Should we either `?? ''` to ensure a string, or use a `Title not available` message, or skip things with no title data?

Reply:
I think this `title` field has the same odds of being null at the `$instance->title` it replaces, but if it actually happens, I think it's better to handle it in the template using your suggestion of `Title not available`.  This is done.  Empty string is not great since the text will become a link.

Reply:
That makes sense! I've added a note to the driver spec indicating that title may be null. I'll finish up the rest of this review when I'm back in the office next week.